code/modules/interaction_tests/test_puppeteer.dm

```diff
@@ -0,0 +1,49 @@
+/datum/test_puppeteer
+	var/mob/living/carbon/puppet
+	var/list/tracked_atoms
```

Comment:
IMO, tracking the atoms should be left to the unit test, not the puppet

Reply:
Done with the new allocator.